### PR TITLE
bugfix/unjail-from-inactive

### PR DESCRIPTION
**Pull request by sasurobert**
unjail from inactive bugfix. temp rating must be set to unJailRating.

### CODE REVIEW

**Review by raduchis — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]
This looks like breaking changes. We might get along with this one on the mainnet.

File: process/peer/process.go [hunk 1]
```diff
@@ -75,6 +75,7 @@ type validatorStatistics struct {
 	jailedEnableEpoch               uint32
 	belowSignedThresholdEnableEpoch uint32
 	flagJailedEnabled               atomic.Flag
+	flagInactiveToJailCorrection    atomic.Flag
```
Comment (SebastianMarian) [inline process/peer/process.go:78]: flagInactiveToJailCorrectionEnabled

File: process/peer/process.go [hunk 2]
```diff
@@ -1195,5 +1196,6 @@ func (vs *validatorStatistics) LastFinalizedRootHash() []byte {
 // EpochConfirmed is called whenever a new epoch is confirmed
 func (vs *validatorStatistics) EpochConfirmed(epoch uint32) {
 	vs.flagJailedEnabled.Toggle(epoch >= vs.jailedEnableEpoch)
+	vs.flagInactiveToJailCorrection.Toggle(epoch == vs.jailedEnableEpoch)
```
Comment (SebastianMarian) [inline process/peer/process.go:1199]: Should not be >= ?

File: process/peer/process.go [hunk 3]
```diff
@@ -1195,5 +1196,6 @@ func (vs *validatorStatistics) LastFinalizedRootHash() []byte {
...
 	log.Debug("validatorStatistics: jailed", "enabled", vs.flagJailedEnabled.IsSet())
```
Comment (SebastianMarian) [inline process/peer/process.go:1200]: Add also this log line: log.Debug("validatorStatistics: inactiveToJailCorrection", "enabled", vs.flagInactiveToJailCorrectionEnabled.IsSet())

File: process/peer/process.go (hunk 1 above)
Comment (SebastianMarian) [inline process/peer/process.go:78]: flagJailedEnabled is not used anymore

File: process/peer/process.go (hunk 2 above)
Comment (sasurobert) [reply process/peer/process.go:1199]: no - this should be done only once. In the epoch when the activation happens. we have a case with the old code - before flag activation - peer.UnStakedEpoch == maxuint32 && peer.List == Inactive means that it is jailed..

File: process/peer/process.go (hunk 1 above)
Comment (sasurobert) [reply process/peer/process.go:78]: renamed

File: process/peer/process.go (hunk 3 above)
Comment (sasurobert) [reply process/peer/process.go:1200]: done

File: process/peer/process.go (hunk 1 above)
Comment (sasurobert) [reply process/peer/process.go:78]: it is used in PeerAccountToValidatorInfo

File: process/peer/process.go (hunk 2 above)
Comment (SebastianMarian) [reply process/peer/process.go]: ok

File: process/peer/process.go (hunk 1 above)
Comment (SebastianMarian) [reply process/peer/process.go]: ok

**Review by SebastianMarian — DISMISSED** [review]

**Review by iulianpascalau — DISMISSED** [review]

File: process/peer/process.go [hunk 4]
```diff
@@ -1195,5 +1197,7 @@ func (vs *validatorStatistics) LastFinalizedRootHash() []byte {
 // EpochConfirmed is called whenever a new epoch is confirmed
 func (vs *validatorStatistics) EpochConfirmed(epoch uint32) {
 	vs.flagJailedEnabled.Toggle(epoch >= vs.jailedEnableEpoch)
+	vs.flagInactiveToJailEnabled.Toggle(epoch == vs.jailedEnableEpoch)
```
Comment (raduchis) [inline process/peer/process.go:1200]: is this flag needed?

File: process/peer/process.go [hunk 5]
```diff
@@ -215,7 +216,7 @@ func (vs *validatorStatistics) saveUpdatesForList(
 		}
 
 		isNodeLeaving := (peerType == core.WaitingList || peerType == core.EligibleList) && peerAcc.GetList() == string(core.LeavingList)
-		isNodeJailed := vs.flagJailedEnabled.IsSet() && peerType == core.InactiveList && peerAcc.GetUnStakedEpoch() == core.DefaultUnstakedEpoch
+		isNodeJailed := vs.flagInactiveToJailEnabled.IsSet() && peerType == core.InactiveList && peerAcc.GetUnStakedEpoch() == core.DefaultUnstakedEpoch
```
Comment (raduchis) [inline process/peer/process.go:219]: this is going to be true exactly in the jailedEnabledEpoch, is this the expected behavior?

**Review by raduchis — APPROVED** [review]

File: process/peer/process.go (hunk 4 above)
Comment (sasurobert) [reply process/peer/process.go:1200]: yes. it is needed.

File: process/peer/process.go (hunk 5 above)
Comment (sasurobert) [reply process/peer/process.go:219]: yes. the changes must happen only once.

**Review by LucianMincu — DISMISSED** [review]
System tests passed.

File: process/peer/process.go [hunk 6]
```diff
@@ -444,6 +445,7 @@ func getActualList(peerAccount state.PeerAccountHandler) string {
 	savedList := peerAccount.GetList()
 	if peerAccount.GetUnStakedEpoch() == core.DefaultUnstakedEpoch {
 		if savedList == string(core.InactiveList) {
+			log.Debug("getActual list returns jailed")
```
Comment (SebastianMarian) [inline process/peer/process.go:447]: This log was not removed

**Review by SebastianMarian — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**